src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -1014,8 +1014,15 @@ private static BoundExpression NullableAlwaysHasValue(BoundExpression expression
                 }
             }
 
+            // do not rewrite user defined conversion 
+            // - in expression trees or 
+            // - special situations when converting to ReadOnlySpan, which has special support in codegen
+            var doNotLowerToCall = _inExpressionLambda ||
+                                   (rewrittenOperand.Type.IsArray()) &&
```

Comment:
Nit: perhaps consolidate the && case on one line?
Also update the comment "- when converting from array to ReadOnlySpan, which has ..."